### ISSUE TITLE
Using core.simd with DMD when possible

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/intel-intrinsics#59]
- [ ] enable core.simd when D_SIMD is defined (not for DMD 32-bit target)
- [x] MMX-size registers still have to be emulated tough

(EDIT: hehe, a few DMD bugs have to be reported and workarounded in the process... this will give future work)

**Comment by p0nce** [comment]
3 Bugzilla entry later:
- https://issues.dlang.org/show_bug.cgi?id=21469
- https://issues.dlang.org/show_bug.cgi?id=21470
- https://issues.dlang.org/show_bug.cgi?id=21474

Blocked by the third.
That last one bug is a bit harder to workaround, reducing takes time and isn't single site.

**Comment by p0nce** [comment]
Blocked by https://issues.dlang.org/show_bug.cgi?id=21474

Issues with this bug are hard to workaround and I don't know if there is more than 2 occurences (likely). This will also crop up in code using intel-intrinsics.

**Comment by p0nce** [comment]
After lobbying was fixed in DMD master, so we can proceed for __VERSION__ >= 2095 for DMD

**Comment by p0nce** [comment]
Enabled for __VERSION__ >= 2096
